skimage/measure/_blur_effect.py

```diff
@@ -85,6 +85,8 @@ def blur_effect(image, h_size=11, channel_axis=None, reduce_func=np.max):
         T = np.maximum(0, im_sharp - im_blur)
         M1 = np.sum(im_sharp[slices])
         M2 = np.sum(T[slices])
+        if M1 == 0:
+            raise ValueError('Division by zero will cause NaN')
```

Comment:
```suggestion
            raise ZeroDivisionError
```
Also, you could move this check one line up!